### PR TITLE
Method signatures for Rust layout objects

### LINKED ISSUE #2672: RustLayout class and its methods lack signatures in the mock file
`LayoutObj::obj_type` should have doc comments and their output should go to `trezorui2.pyi`

### PR DESCRIPTION
**Pull request by matejcik**
some preliminary work for the global layout PR.
fixes #2672
also converts most functions in the layout files to `def -> Awaitable` format (instead of `async def -> T` with a single "return await").

### CODE REVIEW

**Comment by matejcik** [comment]
> What is the benefit of this change?

one less allocation per called function.
every async function instantiates a generator object which is then awaited. a plain function just gets called without this overhead.

File: core/src/trezor/wire/context.py [hunk 1]
```diff
@@ -159,7 +161,7 @@ async def write(self, msg: protobuf.MessageType) -> None:
 CURRENT_CONTEXT: Context | None = None
 
 
-def wait(*tasks: Awaitable) -> Any:
+def wait(task: Awaitable[T]) -> Awaitable[T]:
     """
     Wait until one of the passed tasks finishes, and return the result, while servicing
```
Comment (mmilata) [inline core/src/trezor/wire/context.py:166]: Docstring still refers to multiple tasks.

**Review by mmilata — APPROVED** [review]
Very nice, the upy<->rs interface was a big untyped mess ...

File: core/src/trezor/wire/context.py (hunk 1 above)
Comment (matejcik) [reply core/src/trezor/wire/context.py]: 9c630f56d00161e7d5209036428d5228006c87b2

File: core/embed/rust/src/ui/model_tr/layout.rs [hunk 2]
```diff
@@ -2054,21 +2054,22 @@ pub static mp_module_trezorui2: Module = obj_module! {
...
-    /// ) -> object:
+    /// ) -> LayoutObj[UiResult]:
     ///    """Device recovery homescreen."""
     Qstr::MP_QSTR_confirm_recovery => obj_fn_kw!(0, new_confirm_recovery).as_obj(),
 
     /// def select_word_count(
     ///     *,
     ///     dry_run: bool,  # unused on TR
     /// ) -> int | str:  # TR returns str
+    /// ) -> LayoutObj[int | str]:
```
Comment (grdddj) [inline core/embed/rust/src/ui/model_tr/layout.rs:2065]: Seems we should delete the line above to compensate for the new line

Comment (matejcik) [reply core/embed/rust/src/ui/model_tr/layout.rs:2065]: good catch, and how tf did this not completely break typechecking?

Comment (matejcik) [reply core/embed/rust/src/ui/model_tr/layout.rs:2064]: b9e1bc8fa4e1ffbe1ce5c7142fd3ec417a86a7f0

**Pull request merged**